src/Sylius/Component/Order/Model/Order.php

```diff
@@ -435,7 +435,9 @@ public function getAdjustmentsTotal($type = null)
 
         $total = 0;
         foreach ($this->getAdjustments($type) as $adjustment) {
-            $total += $adjustment->getAmount();
+            if (!$adjustment->isNeutral()) {
+                $total += $adjustment->getAmount();
+            }
```

Comment:
I think the problem here (and you have already mentioned it) is the lack of clarity around adjustments at the `Order` level.

From my perspective, these can **only** be related to shipping. Any other adjustments would then be associated eventually with `OrderItemUnit`s. 

If you accept this then your use-case above isn't required, because you don't need to do:

```
$shippingTotal = $this->getAdjustmentsTotal(AdjustmentInterface::SHIPPING_ADJUSTMENT);
$shippingTotal += $this->getAdjustmentsTotal(AdjustmentInterface::ORDER_PROMOTION_ADJUSTMENT);
$shippingTotal += $this->getAdjustmentsTotal(AdjustmentInterface::TAX_ADJUSTMENT); 
```

You just do:

```
$this->getAdjustmentsTotal()
```

Because this is the neutral-adjustment total for Shipping.
It's the same for `OrderItem` and `OrderItemUnit` where you want their totals then you want them neutrally adjusted. It's just perhaps unfortunate that Shipping adjustments are on the `Order` object.

If you reject this rule of only Shipping adjustments directly on an order then I agree that `getNonNeutralAdjustmentsTotal` pattern is probably the solution.